packages/api/lib/bucketMapUtils.js

```diff
@@ -36,10 +36,15 @@ async function getYamlFile(bucket, key) {
  */
 const getBucketMap = () => getYamlFile(process.env.system_bucket, process.env.BUCKET_MAP_FILE);
 
-function prependBucketname(name) {
+const prependBucketName = (name) => {
   const prefix = process.env.BUCKETNAME_PREFIX || '';
   return `${prefix}${name}`;
-}
+};
+
+const removePrefixBucketName = (name) => {
+  const prefix = process.env.BUCKETNAME_PREFIX || '';
+  return name.replace(new RegExp(`^${prefix}`), '');
```

Comment:
Super-nit: consider returning name here instead of invoking .replace if !prefix.